x/nft/keeper/grpc_query.go

```diff
@@ -0,0 +1,191 @@
+package keeper
+
+import (
+	"context"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/query"
+	"github.com/cosmos/cosmos-sdk/x/nft"
+)
+
+var _ nft.QueryServer = Keeper{}
+
+// Balance return the number of NFTs of a given class owned by the owner, same as balanceOf in ERC721
+func (k Keeper) Balance(goCtx context.Context, request *nft.QueryBalanceRequest) (*nft.QueryBalanceResponse, error) {
+	if request == nil {
+		return nil, status.Errorf(codes.InvalidArgument, "empty request")
```

Comment:
I like the idea of using grpc standard codes. But in other parts of the SDK we are using sdk errors, eg:
```
errors.ErrInvalidRequest.Wrap("empty request")
```
The motivation to continue using the SDK errors is to have consistent checks. 

what do you think?

Reply:
Okay. At that time, I saw that other modules also use grpc standard codes, so I did the same. Will the SDK error codes be used to unify them in the future?

Reply:
I’m not really sure what the decision is but honestly I think for queries grpc error codes make sense

Reply:
We are converting all errors to SDK errors for queries: https://github.com/cosmos/cosmos-sdk/blob/master/baseapp/abci.go#L584
I think promoting the sdk errors makes more sense because we have better clarity in that.